docs/querying/json.mdx

```diff
@@ -288,78 +292,221 @@ User.create({
+User.create({
+  jsonAttribute: {
+    name: null,
+  },
+});
+```
 
-This means that the following query will check that the `name` property exists and is set to the JSON `null` value:
+```sql
+INSERT INTO "Users" ("jsonAttribute") VALUES ('{"name":null}');
+```
+
+### Querying null values
+
+When comparing against a JSON column, Sequelize will require you to be explicit about whether you wish to compare against the JSON `null` value,
+or the SQL `NULL` value.
+
+#### How to compare against either JSON `null` or SQL `NULL`
+
+Comparing against the JSON `null` value is done by using the `JSON_NULL` constant, or by using the `Op.eq` operator.
+Both of the following queries are equivalent:
 
 ```ts
 User.findAll({
   where: {
-    'jsonAttribute.name': null,
+    jsonAttribute: JSON_NULL,
+  },
+});
+
+User.findAll({
+  where: {
+    jsonAttribute: { [Op.eq]: null },
   },
 });
 ```
 
 ```sql
--- postgres
-"jsonAttribute"->'name' = 'null';
-
--- mysql & friends
-JSON_EXTRACT(`jsonAttribute`, '$.name') = 'null';
+SELECT * FROM "Users" WHERE "jsonAttribute" = 'null';
 ```
 
-If you want to check that something is the SQL `NULL`, you have to use the `IS NULL` operator instead of the `=` operator:
+Comparing against the SQL `NULL` value is done by using the `SQL_NULL` constant, or by using the `Op.is` operator.
+Both of the following queries are equivalent:
 
 ```ts
 User.findAll({
   where: {
-    'jsonAttribute.name': { [Op.is]: null },
+    jsonAttribute: SQL_NULL,
+  },
+});
+
+User.findAll({
+  where: {
+    jsonAttribute: { [Op.is]: null },
   },
 });
 ```
 
 ```sql
 -- postgres
-"jsonAttribute"->'name' IS NULL;
-
--- mysql & friends
-JSON_EXTRACT(`jsonAttribute`, '$.name') IS NULL;
+SELECT * FROM "Users" WHERE "jsonAttribute" IS NULL;
 ```
 
-If you want to check that something is either the JSON `null`, or the SQL `NULL`, you must use both operators:
+You can of course combine the two:
 
 ```ts
+import { or, SQL_NULL, JSON_NULL } from '@sequelize/core';
+
 User.findAll({
-  where: or(
-    { 'jsonAttribute.name': { [Op.eq]: null } },
-    { 'jsonAttribute.name': { [Op.is]: null } },
-  ),
+  where: {
+    jsonAttribute: or(SQL_NULL, JSON_NULL),
+  },
 });
 ```
 
 ```sql
--- postgres
-   "jsonAttribute"->'name' = 'null'
-OR "jsonAttribute"->'name' IS NULL;
+SELECT * FROM "Users" WHERE "jsonAttribute" IS NULL OR "jsonAttribute" = 'null';
+```
 
--- mysql & friends
-   JSON_EXTRACT(`jsonAttribute`, '$.name') = 'null'
-OR JSON_EXTRACT(`jsonAttribute`, '$.name') IS NULL;
+Both can also be used on nested properties, read further down for when this can be useful.
+
+```ts
+import { or, SQL_NULL, JSON_NULL } from '@sequelize/core';
+
+User.findAll({
+  where: {
+    'jsonAttribute.name': or(SQL_NULL, JSON_NULL),
+  },
+});
 ```
+
+```sql
+SELECT * FROM "Users" WHERE "jsonAttribute"->'name' IS NULL OR "jsonAttribute"->'name' = 'null';
```

Comment:
Should the 'mysql & friends' example remain here?

Reply:
Added!